### PR TITLE
Server

### PR DESCRIPTION
**Pull request by ilydnic**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,11 +3,32 @@
...
+    "passport": "^0.3.2",
+    "passport-facebook": "^2.1.1",
+    "passport-google-oauth": "^1.0.0",
+    "passport-local": "^1.0.0",
+    "passport-twitter": "^1.0.4",
+    "path": "^0.12.7",
+    "sequelize": "^3.23.4",
+    "serve-favicon": "^2.3.0",
+    "socket.io": "^1.4.8",
+    "util": "^0.10.3"
```
Comment (ilydnic) [inline package.json:31]: careful, comma at the end of this line

File: server/app/routes/days.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    let whereCondition = {};
+    if (req.requestedUser) {
+        whereCondition.where = {};
+        whereCondition.where.userId = req.requestedUser.id;
+    }
+    Day.all(whereCondition)
+    .then(days => {
+        res.json(days);
+    });
+});
```
Comment (ilydnic) [inline server/app/routes/days.js:19]: error handling with next

File: server/app/routes/days.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    })
+    .catch(next);
+});
+
+router.param('dayId', function(req, res, next, userId) {
+    Day.findById(dayId)
+    .then(day => {
+        if (!day) {
+            res.status(404);
+            throw next(new Error('Day not found.'));
```
Comment (ilydnic) [inline server/app/routes/days.js:43]: you don't need to throw next for it to error handle

File: server/app/routes/locations.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        res.json(location);
+    })
+    .catch(next);
+});
+
+router.param('locationId', function(req, res, next, userId) {
+    Location.findById(locationId)
+    .then(location => {
+        if (!location) {
+            res.status(404);
```
Comment (ilydnic) [inline server/app/routes/locations.js:43]: setting an error status on res doesn't pass that status to the error handler. you can set the status on the error you pass to next

**Comment by wfa207** [comment]
Looks like we'll be going with react-native-store for now; we'll revisit if we decide to go back with a full back-end

**Pull request closed without merging**